### ISSUE TITLE
Passing custom data via form prop to fields

### ISSUE THREAD

**Issue by dmitryverkhovtsev** [foxhound87/mobx-ajv-form#187]
Hi! I have a form with multiple selects and I don't think that define select options for each field in a React dumb view component is a good idea.

I think its better to pass custom object (selectOptions) to form like this:
```
const fields = [
...
'CreditRating',
...];
const values = {...};
const labels = {...};
const rules = {...};
const selectOptions = {
    ...
    'CreditRating': ['Poor', 'Average', 'Excellent', 'Unsure'],
    ...
}

const form = new myForm({ fields, values, labels, rules, plugins, selectOptions });
```
I read it https://foxhound87.github.io/mobx-react-form/docs/api-reference/extend.html, but still don't catch the idea how to do it:

How extend Form and Field to pass down to Fields custom props or object? (in this case selectOptions)

**Comment by foxhound87** [comment]
You can extend a **CreditRatingField** providing an `options` prop:

```javascript
class CreditRatingField extends Field {

  options = ['Poor', 'Average', 'Excellent', 'Unsure'];

  constructor(data) {
    super(data);
  }
}

class MyForm extends Form {

  makeField(field) {
    switch(field.key) {
      case 'creditRating':
        return new CreditRatingField(field);
      default:
        return new Field(field);
    }
  }
}
```

Into the custom field class, for example, you can write the logic to fetch the options.

then you can use it: 

```javascript 
form.$('creditRating').options
````

**Comment by dmitryverkhovtsev** [comment]
Thank you. 
But this solution lack flexibility.
It's good for 1 or 2 selects but I have near 50 selects. 
Why we cannot pass custom data (as prop) to fields?

As I understand an object passed with constructor call ignored by parser? 
```const form = new myForm({ fields, values, labels, placeholders, rules, plugins, selectOptions });```
makeField() doesn't receive selectOptions object.

**Comment by foxhound87** [comment]
Eventually I can implement an **options** prop to pass in the constructor.

**Comment by dmitryverkhovtsev** [comment]
It would be great

**Comment by foxhound87** [comment]
The **options** key is already in use in the constructor for the form options.
It will take some time because I have to change the current api and release a major update.

**Comment by foxhound87** [comment]
`options` property is now available on the new [1.18 version](https://github.com/foxhound87/mobx-react-form/releases/tag/v1.18.0)
